docs/blog/2019-03-03-new-schema-customization/index.md

```diff
@@ -0,0 +1,309 @@
+}
+```
+
+Gatsby will now know that you want a Date and not override it with a string.
+
+You can specify types for some or all of the fields that you have on the given node type. Gatsby will add missing fields. This behaviour can be controlled with `@infer` and `@dontInfer` directives.
+
+```graphql
+# For this type `name` won't be added
+type AuthorJson implements Node @dontInfer() {
+  birthday: Date
+}
+
+# For this type `name` won't be added but `birthday` won't get a Gatsby resolver for date formatting
+type AuthorJson implements Node @dontInfer(noDefaultResolvers: true) {
+  birthday: Date
+}
+
+
+# For this type both `name` and `birthday` fields will be added. Current default behaviour, but allows one to be explicit about it.
+type AuthorJson implements Node @infer() {
+  id: ID!
+}
+
+# `birthday` will be Date, but we won't add a Gatsby resolver for date formatting
+type AuthorJson implements Node @infer(noDefaultResolvers: true) {
+  id: ID!
+}
+```
+
+## `createResolvers`
+
+This is a similar API to `setFieldsOnGraphQLNodeType` in that it allows to add new fields and resolvers to types. However, this one is run last, so you'd have all the schema available to you in it. It is also possible to extend the `Query` type to add custom root resolvers, which enables a powerful resolver-based approach to querying your data sources. `createResolvers` is called after third-party schemas are merged (eg ones added by `gatsby-source-graphql`), so you can extend those schemas too.
+
+```js
+exports.createResolvers = ({ createResolvers, schema }) => {
+  createResolvers({
+    AuthorJson: {
+      // Modify birthday resolver so that it uses 1970-01-01 as default date
+      birthday: {
+        resolve(source, args, context, info) {
+          // original resolver available as "info.originalResolver"
+          if (Number.isNaN(new Date(source['birthday'])) {
+            return info.originalResolver({
+              ...source,
+              birthday: '1970-01-01',
+            }, args, context, info)
+          } else {
+            return info.originalResolver(source, args, context, info)
+          }
+        },
+      },
+    },
+  })
+}
+```
+
+It's also possible to create new root fields, for example one that will return all author names as strings.
+
+```js
+createResolvers({
+  Query: {
+    allAuthorFullNames: {
+      type: `[String!]!`,
+      resolve(source, args, context, info) {
+        const authors = context.nodeModel.getAllNodes(
+          {
+            type: `AuthorJson`,
+          },
+          { path: context.path }
+        )
+        return authors.map(author => author.name)
+      },
+    },
+  },
+})
+```
+
+Notice the `context.nodeModel`. We expose our internal node storage to the resolvers, so that one can fetch data from there. In addition to lower lever access functions (`getNode`, `getAllNodes`), full node querying is available in `runQuery` .Full documentation of nodeModel is available at TODO.
```

Comment:
```suggestion
Notice the `context.nodeModel`. We expose our internal node storage to the resolvers, so that one can fetch data from there. In addition to lower lever access functions (`getNodeById`, `getAllNodes`), full node querying is available in `runQuery`. Full documentation of `nodeModel` is available at TODO.
```

Reply:
What would be a good place to document NodeModel? I could work on that.

Reply:
Maybe create a new page in docs, somewhere in GraphQL section. Link to it from createTypes/createResolvers actions.